File: my-folder/0198-house-robber/solution.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 1:
            return nums[0]
        robbed = [0]*n 
        robbed[0] = nums[0]
        robbed[1] = nums[1]
        for i in range(2, n):
            if i > 2:
                robbed[i] = nums[i] + max(robbed[i-2], robbed[i-3])
            else:
                robbed[i] = nums[i] + robbed[i-2]

        return max(robbed[-1], robbed[-2])

File: my-folder/0198-house-robber/solution.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
       
        int n = nums.size();
        int a = 0, b = 0;
        for (int i = 0; i< n; i++)
        {
            int temp = a;
            a = max(a, b + nums[i]);
            b = temp;
        }
        return a;
    }
};
